### PR TITLE
#141/UI/logic/user dogs

### PR DESCRIPTION
**Pull request by pierwszazlewej**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -63,10 +68,11 @@ function App() {
             path="contests/:contestId/classes/:classId/:dogId/dogSummary"
             element={<DogSummaryPage />}
           />
+          <Route path="UserDogPage" element={<UserDogPage />}></Route>
           <Route path="dogData" element={<DogDataPage />} />
           <Route path="contestDetails" element={<ContestDetailsPage />} />
         </Route>
-
+        q
```
Comment (Smyku6) [inline src/App.js:75]: no to q

Comment (pierwszazlewej) [reply src/App.js:75]: q is out!

File: src/Molecules/ClassOrDogButton/ClassOrDogButton.js [hunk 2]
```diff
@@ -56,6 +57,7 @@ ClassOrDogButton.propTypes = {
     exercisesCompleted: PropTypes.number.isRequired,
     exercisesAmount: PropTypes.number.isRequired,
```
Comment (Smyku6) [inline src/Molecules/ClassOrDogButton/ClassOrDogButton.js:58]: remove isRequired from this lines:
```
 exercisesCompleted: PropTypes.number,
    exercisesAmount: PropTypes.number,
```

**Review by Smyku6 — CHANGES_REQUESTED** [review]

File: src/PagesBody/UserDogsPage/UserDogPage.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import MainButton from '../../Atoms/MainButton/MainButton';
+import participants from '../../Data/MongoDBMock/participants';
+import { useNavigate } from 'react-router-dom';
+
+const UserDogPage = () => {
+  const navigate = useNavigate();
+
+  return (
+    <ColumnWrapper paddingLeftRight={1} paddingTop={0.5}>
+      {participants[0].dogs.map((dog, index) => {
```
Comment (Smyku6) [inline src/PagesBody/UserDogsPage/UserDogPage.js:12]: you can change this `participants[0].dogs` to reading data from Context:

image

File: src/PagesBody/UserDogsPage/UserDogPage.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <ColumnWrapper paddingLeftRight={1} paddingTop={0.5}>
+      {participantDogs.map((dog, index) => {
+        const { dogName, dogId } = dog;
+        return (
+          <ClassOrDogButton
+            key={dogId}
+            dogInfo={{
+              index,
+              dogName,
```
Comment (Smyku6) [inline src/PagesBody/UserDogsPage/UserDogPage.js:29]: pass here `dogId` too
our ClassOrDogButton use this props:
image

**Review by Smyku6 — CHANGES_REQUESTED** [review]

File: src/PagesBody/UserDogsPage/UserDogPage.js (hunk 4 above)
Comment (pierwszazlewej) [reply src/PagesBody/UserDogsPage/UserDogPage.js:30]: I've updated a prop, but still struggling with this context usage. Maybe I'm trying to use a wrong file?

**Review by Smyku6 — APPROVED** [review]

**Pull request merged**